src/lib.rs

```diff
@@ -415,6 +422,51 @@ impl<W: Write> Archive<W> {
         self.append(&header, file)
     }
 
+    /// Adds a directory to this archive with the given path as the name of the
+    /// directory in the archive.
+    ///
+    /// This will use `stat` to populate a `Header`, and it will then append the
+    /// directory to the archive with the name `path`.
+    ///
+    /// Note that this will not attempt to seek the archive to a valid position,
+    /// so if the archive is in the middle of a read or some other similar
+    /// operation then this may corrupt the archive.
+    ///
+    /// Also note that after all files have been written to an archive the
+    /// `finish` function needs to be called to finish writing the archive.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fs;
+    /// use tar::Archive;
+    ///
+    /// let mut ar = Archive::new(Vec::new());
+    ///
+    /// // Open the file at one location, but insert it into the archive with a
+    /// // different name.
+    /// let s = fs::metadata(".").unwrap();
+    /// ar.append_dir("bardir", &s).unwrap();
+    /// ```
+    pub fn append_dir<P: AsRef<Path>>(&self, path: P, stat: &fs::Metadata)
```

Comment:
I thought about this - what do you think the method should take? A path?
There's sadly no 'open directory' object.